docs/components/table-data.md

```diff
@@ -0,0 +1,479 @@
+}
+```
+
+:::tip NOTE
+Please note that you have to set [`clientSort` prop](#clientsort) to `true` as well as to pass `sortable: true` and `useSortHandlerFunction: true` in the respective [headers object](#headers) to make a column sortable via `sortHandlerFunction`.
+:::
+
+Please refer to [`fetcher` prop documentation](#fetcher) for example.
+
+### cacheIdentifier
+
+The fetcher functionality makes use of [SWRV](https://docs-swrv.netlify.app/) to handle caching of response data. In order to take advantage of this caching, SWRV needs a way to identify which cache entry is associated with the table.
+
+The identifier should be a string and will default to `''` if not provided. In that scenario, we will generate a random ID for the identifier every time the table is mounted.
+
+:::danger DANGER
+This identifier must be unique across all KTableData instances across the entire Vue app, otherwise there is a risk that [SWRV](https://docs-swrv.netlify.app/) will return the cached data of the wrong table.
+:::
+
+### fetcherCacheKey
+
+Whenever the cache key is changed the fetcher will automatically be called and attempt to fetch new table data.
+
+```vue
+<template>
+  <KTableData
+    cache-identifier="fetcher-cache-key-example-table"
+    :fetcher-cache-key="String(cacheKey)"
+    :fetcher="fetcher"
+    :headers="headers"
+  />
+</template>
+
+<script setup lang="ts">
+const cacheKey = ref<number>(1)
+
+const itemDeleted = (): void => {
+  cacheKey.value++ // triggers refetch or reactivity
+}
+</script>
+```
+
+An alternate implementation is to apply a `key` attribute to the KTableData in conjunction with a `fetcher` and the [SWRV](https://docs-swrv.netlify.app/) `revalidate` function. To prevent unnecessary calls on mount, the `key` `ref` should have an initial value of `0`.
+
+Since the `fetcher` function will handle the initial fetch of the data, we want the cache key for the `revalidate` function to be a falsey value on page load (to avoid an unnecessary duplicate call), and will manually call `revalidate()` and increment the `key` to trigger a refetch and redraw of the table.
+
+```html
+<template>
+  <KTableData :key="key" :fetcher="fetcher" :headers="headers" />
+</template>
+
+<script setup lang="ts">
+const key = ref<string>(0) // initialized to zero
+
+const fetcher = async ({ pageSize, page, query, offset = null }) => {
+  try {
+    const res = await services.getAll(pageSize, offset)
+
+    // handle data
+  } catch (error) {
+    // handle error
+  }
+}
+
+const { revalidate } = composables.useRequest(
+  () => key.value && `service-list-${key.value}`, // will evaluate to falsey on mount, preventing an extra call
+  () => { return fetcher() }
+)
+
+const handleDelete = (id): void => {
+  try {
+    const res = await services.delete(id)
+
+    key.value++
+    revalidate()
+  } catch (error) {
+    // handle error
+  }
+}
+</script>
+```
+
+### searchInput
+
+Pass in a string of search query for server-side table filtering.
```

Comment:
Shouldn't reference "server-side filtering" - needs to be reworded to explain that the searchInput is returned as a param to the fetcher that can then be used to modify the API request